### PR TITLE
Feature/parsing array

### PR DESCRIPTION
**Pull request by o-tsaruk**
Discriminate between mapping and arrays, 
pop, push and length,
different fixes.

### CODE REVIEW

File: examples/contracts/ERC1155/ERC1155/lib.rs [hunk 1]
```diff
@@ -156,21 +156,15 @@ pub mod erc_1155 {
             accounts: Vec<AccountId>,
             ids: Vec<u128>,
         ) -> Result<Vec<u128>, Error> {
-            if accounts.length != ids.length {
+            if accounts.len()? != ids.len()? {
```
Comment (coreggon11) [inline examples/contracts/ERC1155/ERC1155/lib.rs]: len() does not return an error

File: src/assembler.rs [hunk 2]
```diff
@@ -798,6 +798,10 @@ impl ToTokens for Statement {
                     #left #operation #right;
                 })
             }
+            Statement::ArrayMethodCall(variable, method_str, element) => {
```
Comment (coreggon11) [inline src/assembler.rs]:
```suggestion
            Statement::ArrayFunctionCall(variable, function_raw, element) => {
```

File: src/parser.rs [hunk 3]
```diff
@@ -330,6 +330,12 @@ lazy_static! {
     static ref REGEX_BREAK:Regex = Regex::new(
         r#"^\s*break\s*;"#,
     ).unwrap();
+    static ref REGEX_ARRAY_METHOD:Regex = Regex::new(
```
Comment (coreggon11) [inline src/parser.rs]:
```suggestion
    static ref REGEX_ARRAY_FUNCTION:Regex = Regex::new(
```

File: src/parser.rs [hunk 4]
```diff
@@ -266,7 +266,7 @@ lazy_static! {
     .unwrap();
     static ref REGEX_ASSIGN: Regex = Regex::new(
         r#"(?x)
-        ^\s*(?P<left>[0-9a-zA-Z_\[\].]+?)\s*
+        ^\s*(?P<left>[0-9a-zA-Z_\[\].+\-*/]+?)\s*
```
Comment (coreggon11) [inline src/parser.rs:291]: why add these chars?

Comment (o-tsaruk) [reply src/parser.rs:291]: when we have expression in index
`array[element+2] = 0;`

**Pull request merged**